app/models/create_expenses_with_receipt.rb

```diff
@@ -0,0 +1,29 @@
+class CreateExpensesWithReceipt
+  def initialize(expense_params)
+    @params = expense_params
+  end
+
+  def create
+    return false unless picture_exists?
+
+    Expense.transaction do
+      create_expense
+      create_receipt
+    end
+    @expense
+  end
```

Comment:
Yes, it should. What happens when a transaction fails? Does it blow up, or it just rolls back and fails silently?

You could assume that if there is a persisted expense then it was successful. In that case, you just return `@expense.persisted?`.